src/store/MockLocalStorageDependency.ts

```diff
@@ -0,0 +1,24 @@
+import {LocalStorageDependency} from './LocalStorageStore';
+
+export class MockLocalStorageDependency<T extends Record<string, object>> implements LocalStorageDependency {
+    public currentData: T;
+
+    constructor(data: T) {
+        this.currentData = data;
+    }
+
+    getItem(key: string): string | null {
+        if (this.currentData[key]) {
+            return JSON.stringify(this.currentData[key]);
+        }
+
+        return null;
+    }
+
+    setItem(key: string, value: string): void {
+        this.currentData[key as keyof T] = JSON.parse(value);
+    }
```

Comment:
This file won't need to change unless we want to call more methods on `localStorage`. It essentially mocks `localStorage` itself, at least the subset of its methods being used by the application